docs/user-guide/bandwidth-resources.md

```diff
@@ -0,0 +1,171 @@
+decisions about which nodes to place pods on, and contention for resources can be handled in a
+consistent manner.
+
+*ingress* and *egress* bandwidth are each a *resource type*.  All resource types have a base unit.
+For both ingress and egress this unit is bits per second.
+
+Ingress and egress are collectively referred to as *bandwidth resources*.  Bandwidth
+resources are measureable quantities which can be requested, allocated, and consumed.  They are
+distinct from [API resources](working-with-resources.md).  API resources, such as pods and
+[services](services.md) are objects that can be written to and retrieved from the Kubernetes API
+server.
+
+## Pod Resource Limits
+
+Each Pod can optionally specify `spec.containers[].resources.limits.ingress` and/or
+`spec.containers[].resources.limits.egress`.  The `spec.resources.requests` field is not
+currently used and need not be set.
+
+Specifying resource limits is optional.  In some clusters, an unset value may be replaced with a
+default value when a pod is created or updated.  The default value depends on how the cluster is
+configured.
+
+Although limits can only be specified on individual containers, it is convenient to talk about pod
+resource limits.  A *pod resource limit* for a particular resource type is the sum of the resource
+limits of that type for each container in the pod, with unset values treated as zero.
+
+The following pod has two containers.  Each has a limit of 1Mbit of ingress bandwidth and 2Mbit of
+egress bandwidth.
+The pod can be said to have a limit of 2Mbit of ingress and 4Mb of egress.
+
+*Note* Because of limitations in container runtimes, we can't actually limit the bandwidth between
+containers, so different containers in a Pod may steal from each other.  We can (and do) limit
+the overall bandwidth for the Pod.  We hope to refine this as capabilities for hierarchical
+network namespaces come into the container runtimes like Docker.
+
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: frontend
+spec:
+  containers:
+  - name: db
+    image: mysql
+    resources:
+      limits:
+        ingress: "1Mi"
+        egress: "2Mi"
+  - name: wp
+    image: wordpress
+    resources:
+      limits:
+        ingress: "1Mi"
+        egress: "2Mi"
+```
+
+## How Pods with Resource Limits are Scheduled
+
+When a pod is created, the Kubernetes scheduler selects a node for the pod to
+run on.  Each node has a maximum capacity for each of the resource types: the
+amount of CPU and memory it can provide for pods.  The scheduler ensures that,
+for each bandwidth resource type (ingress/egress), the sum of the resource limits of the
+containers scheduled to the node is less than the capacity of the node.  Note
+that although actual bandwidth resource usage on nodes is very low, the
+scheduler will still refuse to place pods onto nodes if the capacity check
+fails.  This protects against a resource shortage on a node when resource usage
```

Comment:
The Resource QoS proposal supports 3 tiers: Guaranteed, Burstable, and BestEffort. #11713

a) I want QoS to go in before this.
b) We should figure out what we want to do with network for those 3 tiers.

cc @AnanyaKumar @vmarmol @vishh @dchen1107 @thockin